crates/ethereum/evm/src/execute.rs

```diff
@@ -102,14 +95,14 @@ where
         self.eth_executor(db)
     }
 
-    fn batch_executor<DB>(&self, db: DB) -> Self::BatchExecutor<DB>
+    fn batch_executor<DB>(&self, db: DB, prune_modes: PruneModes) -> Self::BatchExecutor<DB>
```

Comment:
I think I preferred it as a field, but up to you

Reply:
I tried this first but the problem is that this is updated in the execution stage